### ISSUE TITLE
`clippy::panic`/`clippy::todo`/etc dont trigger when direct argument to macros.

### ISSUE THREAD

**Issue by vivax3794** [rust-lang/rust-clippy#17699]
### Summary

Seems like the lints against panic-y macros dont trigger when used directly in macros.

### Lint Name

clippy::panic

### Reproducer

I tried this code:

```rust
#![deny(clippy::todo)]

pub fn main() {
    println!("{}", todo!("test"));
}
```

I expected to see this happen: Get a error on the use of `todo!`

Instead, this happened: only got a `clippy::diverging_sub_expression` trigger.

if I wrap the `todo!` in a function call like this the `clippy::todo` lint fires fine:
```rust
#![deny(clippy::todo)]

fn foo(x: &str) -> &str {x}

pub fn main() {
    println!("{}", foo(todo!("test")));
}
```

### Version

```text
rustc 1.100.0-nightly (f248f4038 2026-09-05)
binary: rustc
commit-hash: f248f4038796913873f11ca65b1b901e311c8dae
commit-date: 2026-09-05
host: x86_64-unknown-linux-gnu
release: 1.100.0-nightly
LLVM version: 23.1.1
```

**Comment by qdot3** [comment]
It is also [warned by built-in `unreachable_code` lint on by default](https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=e38158db74fd092ae960f64850b4c557) and then is likely to be removed from production code. So, adding `known problem` section might be sufficient.

**Comment by vivax3794** [comment]
> It is also [warned by built-in `unreachable_code` lint on by default](https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=e38158db74fd092ae960f64850b4c557) and then is likely to be removed from production code. So, adding `known problem` section might be sufficient.

I kinda disagree, for example I have unreachable_code as a warn, but todos as a deny, so the difference is important to me. While yes I am likely to remove both before a push, it still seems like a bug that should be fixed as its pretty unexpected and slightly confusing.

**Comment by vivax3794** [comment]
in fact not all cases of this causes `unreachable_code`
```rust
#![deny(clippy::todo)]

macro_rules! foo {
    ($end:expr) => {{
        $end
    };}
}

fn main() {
    foo!(todo!("test"));
}
```

```
    Checking playground v0.0.1 (/playground)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.46s
```

**Comment by qdot3** [comment]
I was wrong.

@rustbot claim
